web/src/views/DandisetLandingView/DandisetLandingView.vue

```diff
@@ -152,30 +152,63 @@ export default {
       girderDandiset: (state) => state.girderDandiset,
       publishDandiset: (state) => state.publishDandiset,
       loading: (state) => state.loading,
+      dandisetVersions: (state) => state.versions,
     }),
   },
   watch: {
     identifier: {
       immediate: true,
       async handler(identifier) {
-        this.$store.dispatch('dandiset/fetchGirderDandiset', { identifier });
-      },
-    },
-    girderDandiset: {
-      immediate: true,
-      async handler(value) {
         const { version } = this;
-        if (value && isPublishedVersion(version)) {
-          const { _id: girderId, meta: { dandiset: { identifier } } } = value;
-          this.$store.dispatch('dandiset/fetchPublishDandiset', { identifier, girderId, version });
-        }
+        this.$store.dispatch('dandiset/initializeDandisets', { identifier, version });
       },
     },
+    async version(version) {
+      // On version change, fetch the new dandiset (not initial)
+      const { identifier } = this;
+      await this.$store.dispatch('dandiset/fetchPublishDandiset', { identifier, version });
+
+      // If the above await call didn't result in publishDandiset being set
+      if (!this.publishDandiset && version !== draftVersion) { this.navigateToLatestVersion(); }
+    },
+    dandisetVersions() {
+      this.navigateToLatestVersion();
+    },
   },
   methods: {
+    navigateToLatestVersion() {
+      // Set default version to most recent if this dandiset has versions
+      // Otherwise set it to draft
+      let version = draftVersion;
+      const { dandisetVersions: versions } = this;
+
+      if (versions && versions.length) {
+        if (dandisetHasVersion(versions, this.version)) { return; }
```

Comment:
This makes me think that this method doesn't actually `navigateToLatestVersion`. It seems like it only navigates to latest version if the current version isn't valid; if `this.version` is not falsy, then I would assume this method would generally do nothing. Should it be renamed?

Reply:
> if this.version is not falsy, then I would assume this method would generally do nothing

This method only does nothing if `this.version` is contained in `versions`. You're right though, the point of this method isn't to strictly navigate to the latest version, it's to default the version to something known (`draft` or the latest version), if the specified version isn't a known version. I can change the name.

Reply:
Addressed in 7251558.